### PR TITLE
sysrepo BUGFIX segfault with non-existing RPC

### PR DESCRIPTION
**Pull request by syyyr**
Fixes part of #2115 
For the test, I just copied what the other tests did, hopefully it's right

### CODE REVIEW

**Review by jktjkt — APPROVED** [review]

**Comment by michalvasko** [comment]
Uh, sorry, I have fixed it before looking at this PR. So just remove the fix please, the test can be merged.

File: src/sysrepo.c [hunk 1]
```diff
@@ -4202,8 +4202,10 @@ sr_rpc_send(sr_session_ctx_t *session, const char *path, const sr_val_t *input,
 
 cleanup:
     lyd_free_withsiblings(input_tree);
-    while (output_tree->parent) {
-        output_tree = output_tree->parent;
+    if (output_tree) {
+        while (output_tree->parent) {
+            output_tree = output_tree->parent;
+        }
```
Comment (michalvasko) [inline src/sysrepo.c:4208]: Fixed already.

File: tests/test_rpc_action.c [hunk 2]
```diff
@@ -394,6 +394,20 @@ test_rpc(void **state)
...
+    assert_int_equal(ret, SR_ERR_LY);
+    ret = sr_get_error(st->sess, &err_info);
+    assert_int_equal(ret, SR_ERR_OK);
+    assert_int_equal(err_info->err_count, 1);
+    assert_string_equal(err_info->err[0].message, "Schema node not found.");
+    assert_string_equal(err_info->err[0].xpath, "/ops:invalid");
+    assert_null(output);
+    assert_int_equal(output_count, 0);
+
+
```
Comment (michalvasko) [inline tests/test_rpc_action.c:410]: Formatting - extra line.

**Review by michalvasko — CHANGES_REQUESTED** [review]

File: tests/test_rpc_action.c (hunk 2 above)
Comment (syyyr) [reply tests/test_rpc_action.c]: Done

File: src/sysrepo.c (hunk 1 above)
Comment (syyyr) [reply src/sysrepo.c]: Done

**Comment by syyyr** [comment]
All done

**Review by michalvasko — APPROVED** [review]

**Comment by michalvasko** [comment]
Fine, thanks.

**Pull request merged**